test/cadet_web/controllers/user_controller_test.exs

```diff
@@ -69,32 +70,77 @@ defmodule CadetWeb.UserControllerTest do
     end
 
     @tag authenticate: :student
-    test "success, student story skips submitted", %{conn: conn} do
+    test "success, student story skips unopen assessments", %{conn: conn} do
+      build_assessments_starting_at(Timex.shift(Timex.now(), days: 1))
+      build_assessments_starting_at(Timex.shift(Timex.now(), months: -1))
+
+      valid_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: -1))
+
+      for assessment <- valid_assessments do
+        assessment
+        |> Assessment.changeset(%{is_published: false})
+        |> Repo.update!()
+      end
+
+      story =
+        conn
+        |> get("/v1/user")
+        |> json_response(200)
+        |> Map.get("story")
+
+      assert story == nil
+    end
+
+    @tag authenticate: :student
+    test "success, student story does not skip attempting", %{conn: conn} do
+      user = conn.assigns.current_user
+
+      [assessment | _] = build_assessments_starting_at(Timex.shift(Timex.now(), days: -3))
+
+      submission =
+        insert(:submission, %{student: user, assessment: assessment, status: :attempting})
+
+      resp_story =
+        conn
+        |> get("/v1/user")
+        |> json_response(200)
+        |> Map.get("story")
+
+      assert resp_story == assessment.story
+    end
+
+    @tag authenticate: :student
+    test "success, student story skips attempted or submitted", %{conn: conn} do
       user = conn.assigns.current_user
 
-      early_assessments = build_assessments_starting_at(Timex.now())
-      late_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: 1))
+      early_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: -3))
+      late_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: -1))
       assessments_with_indices = Enum.with_index(early_assessments ++ late_assessments)
 
       # Submit for i-th assessment, expect (i+1)th story to be returned
-      for {assessment, idx} <- Enum.drop(assessments_with_indices, -1) do
-        insert(:submission, %{student: user, assessment: assessment})
+      for status <- [:attempted, :submitted] do
+        for {assessment, idx} <- Enum.drop(assessments_with_indices, -1) do
```

Comment:
I think this can be more elegantly achieved using `Enum.chunk_every/2` which is comparable to Ruby's `Enumerable#each_cons`

Reply:
You're right. This is much nicer.